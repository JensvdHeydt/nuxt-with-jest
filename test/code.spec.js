import Vue from 'vue'
import Code from '../components/code'

function getRenderedText (Component, propsData) {
  const Ctor = Vue.extend(Component)
  const vm = new Ctor({ propsData: propsData }).$mount()
  return vm.$el.textContent
}

describe('Code Component', () => {
  let $mounted

  beforeEach(() => {
    $mounted = new Vue(Code).$mount()
  })

  test('snapshot', () => {
    let $html = $mounted.$el.outerHTML
    expect($html).toMatchSnapshot()
  })

  test('renders correctly with different props', () => {
    expect(getRenderedText(Code, {
      data: 'const a = 23'
    })).toBe('const a = 23')

    expect(getRenderedText(Code, {
      data: '<h1>Hero Element</h1>'
    })).toBe('<h1>Hero Element</h1>')
  })

  afterEach(() => {
    $mounted = null
  })
})
