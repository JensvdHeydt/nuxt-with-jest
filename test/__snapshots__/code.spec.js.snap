// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Code Component snapshot 1`] = `"<pre></pre>"`;
